File: PaymentManager/paymentValidation.py
import re
from datetime import datetime


def validate_cvc(cvc: str) -> bool:
    if len(cvc) != 3 or not re.match(r"^\d+$", cvc):
        print(f"CVC {cvc}")
        return False
    return True


def validate_expiry_date(expiry_date: str) -> bool:
    if len(expiry_date) != 5 or expiry_date[2] != '/':
        return False
    try:
        expiry_month = int(expiry_date[:2])
        expiry_year = int(expiry_date[3:])
        correct_expiry_year = int(f"20{expiry_year}")
        current_month = datetime.now().month
        current_year = datetime.now().year
        if correct_expiry_year < current_year or (correct_expiry_year == current_year and expiry_month < current_month):
            return False
    except ValueError:
        print("Invalid date format")
        return False
    return True


def validate_card_number(card_number: str) -> bool:
    pattern = re.compile(r"^\d{4}\s\d{4}\s\d{4}\s\d{4}$")
    if not pattern.match(card_number):
        print("Invalid card number format")
        return False
    return True

File: PayPark/app/src/main/java/com/example/paypark/validation/PaymentValidation.kt
import android.os.Build
import androidx.annotation.RequiresApi
import java.time.LocalDate
import java.util.regex.Pattern

class PaymentValidation {

    fun validateCvc(cvc: String): Boolean {
        if (cvc.length != 3 || !cvc.matches(Regex("\\d+"))) {
            return false
        }
        return true
    }

    @RequiresApi(Build.VERSION_CODES.O)
    fun validateExpiryDate(expiryDate: String): Boolean {
        if (expiryDate.length != 5 || expiryDate[2] != '/') {
            return false
        }
        try {
            val expiryMonth = expiryDate.substring(0, 2).toInt()
            val expiryYear = expiryDate.substring(3).toInt()
            val correctExpiryYear = "20$expiryYear"
            val currentMonth = LocalDate.now().monthValue
            val currentYear = LocalDate.now().year
            if (correctExpiryYear.toInt() < currentYear || (expiryYear == currentYear && expiryMonth < currentMonth)) {
                return false
            }
        } catch (e: NumberFormatException) {
            return false
        }
        return true
    }

    fun validateCardNumber(cardNumber: String): Boolean {
        val pattern: Pattern = Pattern.compile("^\\d{4}\\s\\d{4}\\s\\d{4}\\s\\d{4}$")
        return pattern.matcher(cardNumber).matches()
    }
}
